### PR TITLE
Add handler for Shift-F1 to show advanced keyboard shortcuts

### PR DESCRIPTION
**Pull request by cwh42**
hotkey-help
This PR is more like a RFC - although I am quite confident, that is a good solution.

### CODE REVIEW

File: src/YQWizard.cc [hunk 1]
```diff
@@ -1097,6 +1106,49 @@ void YQWizard::showHelp()
...
+        "<dt>Shift-F8</dt>"
+        "<dd>Save logs to</dd>"
+        "<dt>Shift-F4</dt>"
+        "<dd>Accesibility, Switch to vision impaired palette</dd>"
+        "<dt>Ctrl-Shift-Alt-M</dt>"
+        "<dd>Start/Stop macro recorder</dd>"
+        "<dt>Ctrl-Shift-Alt-P</dt>"
+        "<dd>Play macro</dd>"
+        "<dt>Ctrl-Shift-Alt-T</dt>"
+        "<dd>Dump widget tree to y2log</dd>"
```
Comment (jreidinger) [inline src/YQWizard.cc]: we should not reference y2log here, as it is generic yui help text. so maybe better `Dump widget tree to log`

File: src/YQWizard.cc [hunk 2]
```diff
@@ -1097,6 +1106,49 @@ void YQWizard::showHelp()
...
+     **/
+    _qHotkeysText = _(
+        "<h1>Advanced Hotkeys</h1>"
+        "<dl>"
+        "<dt>Ctrl-Alt-Shift-X</dt>"
+        "<dd>open a terminal window (xterm), useful for VNC installs</dd>"
+        "<dt>Shift-F7</dt>"
+        "<dd>Debug level</dd>"
+        "<dt>Ctrl-Shift-D</dt>"
+        "<dd>Debug module write</dd>"
```
Comment (jreidinger) [inline src/YQWizard.cc]: ugh? what it is? I do not get what it do. Better help text needed.

File: src/YQWizard.cc [hunk 3]
```diff
@@ -1097,6 +1106,49 @@ void YQWizard::showHelp()
...
+        "<dt>Print key</dt>"
+        "<dd>Screenshot</dd>"
```
Comment (jreidinger) [inline src/YQWizard.cc]: what screenshot? What it do? Store screenshot somewhere? and where? or open dialog where to store screenshot?

File: src/YQWizard.cc [hunk 4]
```diff
@@ -1097,6 +1106,49 @@ void YQWizard::showHelp()
...
+    _qHotkeysText = _(
+        "<h1>Advanced Hotkeys</h1>"
+        "<dl>"
+        "<dt>Ctrl-Alt-Shift-X</dt>"
+        "<dd>open a terminal window (xterm), useful for VNC installs</dd>"
+        "<dt>Shift-F7</dt>"
+        "<dd>Debug level</dd>"
+        "<dt>Ctrl-Shift-D</dt>"
+        "<dd>Debug module write</dd>"
+        "<dt>Print key</dt>"
```
Comment (jreidinger) [inline src/YQWizard.cc]: in fact I think key is "PrtScn" or Print Screen key. As print key have some keyboard to fast print of document on printer.

File: src/YQWizard.cc [hunk 5]
```diff
@@ -1097,6 +1106,49 @@ void YQWizard::showHelp()
...
+        "<dt>Ctrl-Shift-Alt-M</dt>"
+        "<dd>Start/Stop macro recorder</dd>"
+        "<dt>Ctrl-Shift-Alt-P</dt>"
+        "<dd>Play macro</dd>"
+        "<dt>Ctrl-Shift-Alt-T</dt>"
+        "<dd>Dump widget tree to y2log</dd>"
+        "<dt>Ctrl-Shift-Alt-Y</dt>"
+        "<dd>Display widget tree browser</dd>"
+        "<dt>Ctrl-Shift-Alt-S</dt>"
+        "<dd>Style Sheet Editor</dd>"
```
Comment (jreidinger) [inline src/YQWizard.cc]: to be consistent it should be Display Style Sheet Editor

Comment (jreidinger) [inline src/YQWizard.cc]: and also other do not have all first letter upper case. It should be unified.

File: package/libyui-qt.changes [hunk 6]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Wed Nov  9 16:56:49 UTC 2016 - cwh@suse.com
+
+- Add handler for Shift-F1 to show advanced keyboard shortcuts
+- 2.46.30
```
Comment (jreidinger) [inline package/libyui-qt.changes]: number do not match reality :wink:

**Comment by jreidinger** [comment]
In general I like this PR. Only major think is if we can use in ncurses shift+F1 and if not, I think we should use same shortcut in ncurses and in qt.

File: src/YQWizard.cc (hunk 1 above)
Comment (cwh42) [inline src/YQWizard.cc]: Speaking about all the comments about the help text I used: It is a direct copy&paste from
https://en.opensuse.org/SDB:YaST_tricks#YaST_Hotkeys
So let's fix it there first - then I can take it back to here.

File: VERSION.cmake [hunk 7]
```diff
@@ -1,6 +1,6 @@
 SET(VERSION_MAJOR "2")
 SET(VERSION_MINOR "46")
-SET(VERSION_PATCH "30")
+SET(VERSION_PATCH "31")
```
Comment (cwh42) [inline VERSION.cmake]: This is what rake version:bump did. It it did that wrong we should also fix it.

File: package/libyui-qt.changes (hunk 6 above)
Comment (cwh42) [inline package/libyui-qt.changes]: Indeed - what ever I did here.

File: src/YQWizard.cc (hunk 1 above)
Comment (jreidinger) [inline src/YQWizard.cc]: yes, good opportunity to also make it clear there

File: VERSION.cmake (hunk 7 above)
Comment (jreidinger) [inline VERSION.cmake]: well, version:bump do common default action, which is increase of patch level. But not all changes fir into patch number increase. I worry we do not change rake version:bump, as detection if API or ABI or none change is too difficult.

**Comment by cwh42** [comment]
Updated PR - changed wording, added bugzilla number and changed versioning.

**Comment by jreidinger** [comment]
LGTM, only remaining question is if we can have same shortcut in ncurses as we should use same shortcut in all backends.

**Comment by cwh42** [comment]
That's what I'll find out today. I let this un-merged until I know more.

**Comment by cwh42** [comment]
hotkey-help-qt
Final layout.

**Pull request merged**